### DISCUSSION TITLE
Exclude arm64 in multiarch using tolerations

### DISCUSSION THREAD

**Discussion by cmwatts1974** [open-policy-agent/community#275] [Community]
Hello looking for the correct way to exclude deployment on nodes of cpu type arm64 in kubernetes using tolerations.  Want to know if the key should be as listed below

 tolerations:
      - key: "arm64"
        operator: "Equal"
        value: "true"
        effect: "NoSchedule"

**Comment by srenatus** [comment]
Please try to provide some context, and your first steps in trying to achieve this:

1. Are you using GK or plain OPA?
2. What is the tolerations feature you're referring to?
3. Can you provide some example input for a positive/negative match? Perhaps in the playground, combined with what you've been trying so far?

Thanks!

**Comment by ngraef** [comment]
This sounds like a Kubernetes question that is not specific to OPA.

I think a `nodeSelector` makes more sense than a toleration in this case. If you only want to run the workload on amd64 nodes, add this to the spec:

```yaml
nodeSelector:
  kubernetes.io/arch: amd64
```

If you do want to run OPA on arm64 nodes, you can use use the multi-arch `openpolicyagent/opa:<version>-static` image variant. If you use this image and still _prefer_ to run the workload on amd64 nodes but want to fall back to arm64 nodes, you can express that using [node affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity) instead of `nodeSelector`.
